UserTools/VtxSeedFineGrid/VtxSeedFineGrid.cpp

```diff
@@ -0,0 +1,470 @@
+		Log("LikelihoodFitterCheck  Tool: Error retrieving RecoDigits,no digit from the RecoEvent!", v_error, verbosity);
+		return false;
+	}
+	auto get_flagsapp = m_data->Stores.at("RecoEvent")->Get("EventFlagApplied",fEventStatusApplied);
+    auto get_flags = m_data->Stores.at("RecoEvent")->Get("EventFlagged",fEventStatusFlagged); 
+    if(!get_flagsapp || !get_flags) {
+      Log("PhaseITreeMaker tool: No Event status applied or flagged bitmask!!", v_error, verbosity);
+      //return false;	
+    }
+    // check if event passes the cut
+    if((fEventStatusFlagged) != 0) {
+      Log("PhaseIITreeMaker Tool: Event was flagged with one of the active cuts.",v_debug, verbosity);
+      return true;	
+    }
+
+	if (useSimpleDir) {
+		Log("Using simple direction", v_debug, verbosity);
+	}
+
+	if (useTrueDir && useMRDTrack) {
+		Log("Unable to use two directions; defaulting to true direction", v_debug, verbosity);	//will change to not use true, to be more general, once MRD track usage is tested and acceptable
+		useMRDTrack = 0;
+	}
+	if (multiGrid) {
+		Log("Using MultiGrid(3)", v_debug, verbosity); //Currently using only three.  May make variable later
+		std::cout << "Using MultiGrid(3)\n";
+	}
+
+	this->FindCenter();
+	if (multiGrid) {
+		for (int i = 0; i < 3; i++) {
+			Center.push_back(vSeedVtxList->at(centerIndex[i]).GetPosition());
+		}
+	}
+	else { Center.push_back(vSeedVtxList->at(centerIndex[0]).GetPosition()); }
+	this->GenerateFineGrid();
+	for (int i = 0; i < Center.size(); i++) {
+		std::cout << "Center " << i << ": " << Center.at(i).X() << ", " << Center.at(i).Y() << ", " << Center.at(i).Z() << endl;
+	}
+	m_data->Stores.at("RecoEvent")->Set("vSeedVtxList", vSeedVtxList, true);
+	Center.clear();
+
+  return true;
+}
+
+
+bool VtxSeedFineGrid::Finalise(){
+	Log("VtxSeedFineGrid exitting", v_debug, verbosity);
+  return true;
+}
+
+Position VtxSeedFineGrid::FindCenter() {
+	double recoVtxX, recoVtxY, recoVtxZ, recoVtxT, recoDirX, recoDirY, recoDirZ;
+	double trueVtxX, trueVtxY, trueVtxZ, trueVtxT, trueDirX, trueDirY, trueDirZ;
+	double seedX, seedY, seedZ, seedT, seedDirX, seedDirY, seedDirZ;
+	double peakX, peakY, peakZ;
+	double bestFOM[3];
+
+	double ConeAngle = Parameters::CherenkovAngle();
+	 
+	// Get true Vertex information
+	Position vtxPos = fTrueVertex->GetPosition();
+	Direction vtxDir = fTrueVertex->GetDirection();
+	trueVtxX = vtxPos.X();
+	trueVtxY = vtxPos.Y();
+	trueVtxZ = vtxPos.Z();
+	trueVtxT = fTrueVertex->GetTime();
+	trueDirX = vtxDir.X();
+	trueDirY = vtxDir.Y();
+	trueDirZ = vtxDir.Z();
+	peakX = trueVtxX;
+	peakY = trueVtxY;
+	peakZ = trueVtxZ;
+	bestFOM[0] = 0; bestFOM[1] = 0; bestFOM[2] = 0;
+	centerIndex[0] = 0; centerIndex[1] = 0; centerIndex[2] = 0;
+	RecoVertex iSeed;
+	RecoVertex thisCenterSeed;
+
+	if (verbosity > 0) cout << "True vertex  = (" << trueVtxX << ", " << trueVtxY << ", " << trueVtxZ << ", " << trueVtxT << ", " << trueDirX << ", " << trueDirY << ", " << trueDirZ << ")" << endl;

```

Comment:
verbosity levels are, in general:
0: error
1: warning
2: message
3+: debug
This does not seem like a warning.